src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -12,325 +13,110 @@ namespace System
 {
     internal static partial class SpanHelpers
     {
-        public static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        public static void ClearWithoutReferences(ref byte b, nuint byteLength)
         {
-            if (byteLength == 0)
-                return;
+            if (!Vector.IsHardwareAccelerated) { goto CannotVectorize; }
+            if (byteLength > (nuint)Vector<byte>.Count) { goto CannotVectorize; }
 
-#if TARGET_AMD64 || TARGET_ARM64 || TARGET_LOONGARCH64
-            // The exact matrix on when ZeroMemory is faster than InitBlockUnaligned is very complex. The factors to consider include
-            // type of hardware and memory alignment. This threshold was chosen as a good balance across different configurations.
-            if (byteLength > 768)
-                goto PInvoke;
```

Comment:
The PInvoke path may be still worth it - to do things like non-temporal stores for very large sizes, etc.